From d834d5277a2512b64576bd867b7095e8f8af5d8a Mon Sep 17 00:00:00 2001
From: yvanliang <yvan6liang@outlook.com>
Date: Fri, 9 Oct 2026 15:40:26 +0800
Subject: [PATCH 04/60] GSI Android 17 compatibility:
 0005-Don-t-crash-if-there-is-IR-HAL-is-not-declared.patch

Change-Id: I4d5b95c25e17892415ba721f8af56aeccd16e403
---
 .../core/java/com/android/server/ConsumerIrService.java   | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/services/core/java/com/android/server/ConsumerIrService.java b/services/core/java/com/android/server/ConsumerIrService.java
index 63dbaa5b3..d68c938ed 100644
--- a/services/core/java/com/android/server/ConsumerIrService.java
+++ b/services/core/java/com/android/server/ConsumerIrService.java
@@ -49,14 +49,6 @@ public class ConsumerIrService extends IConsumerIrService.Stub {
         mWakeLock.setReferenceCounted(true);
 
         mHasNativeHal = getHalService();
-
-        if (mContext.getPackageManager().hasSystemFeature(PackageManager.FEATURE_CONSUMER_IR)) {
-            if (!mHasNativeHal) {
-                throw new RuntimeException("FEATURE_CONSUMER_IR present, but no IR HAL loaded!");
-            }
-        } else if (mHasNativeHal) {
-            throw new RuntimeException("IR HAL present, but FEATURE_CONSUMER_IR is not set!");
-        }
     }
 
     @Override
